lib/patch-series.ts

```diff
@@ -178,42 +180,96 @@ export class PatchSeries {
             basedOn,
             cc,
             coverLetter,
-        } = PatchSeries.parsePullRequestDescription(pullRequestDescription);
+        } = await PatchSeries.parsePullRequest(workDir,
+                                               pullRequestTitle,
+                                               pullRequestBody,
+                                               senderName, senderEmail);
 
         if (basedOn && !revParse(basedOn, workDir)) {
             throw new Error(`Cannot find base branch ${basedOn}`);
         }
 
         const publishToRemote = undefined;
 
-        const project = await ProjectOptions.get(workDir, headCommit, cc || [],
+        const project = await ProjectOptions.get(workDir, headCommit, cc,
                                                  basedOn, publishToRemote,
                                                  baseCommit);
 
-        const wrapCoverLetterAtColumn = 76;
         return new PatchSeries(notes, options, project, metadata,
                                rangeDiff,
-                               md2text(coverLetter, wrapCoverLetterAtColumn),
+                               coverLetter,
                                senderName);
     }
 
-    protected static parsePullRequestDescription(description: string): {
+    protected static async parsePullRequest(workDir: string,
+                                            prTitle: string,
+                                            prBody: string,
+                                            senderName?: string,
+                                            senderEmail?: string):
+    Promise <{
+        coverLetter: string,
+        basedOn?: string,
+        cc: string[],
+    }> {
+        // Replace \r\n with \n to simplify remaining parsing.
+        // Note that md2text() in the end will do the replacement anyway.
+        prBody = prBody.replace(/\r\n/g, "\n");
+
+        // Remove template from description (if template exists)
+        try {
+            let prTemplate =
+                await git(["show",
+                           "upstream/master:.github/PULL_REQUEST_TEMPLATE.md"],
+                          { workDir });
```

Comment:
Not sure why this was moved here from GitGitGadget.  I understand the isolation of the 'mods' to the `prBody` but the function is parse and not cleanup.

Reply:
My biggest reason was to isolate everything that relies upon CRLF/LF in one place, so that I can replace with `\n` and see the consequences in just one place.

I also think that it is well placed: the function parses _cover letter_ out of _pr body_, and removing trash from pr body is reasonable part of that parsing.